src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/LogsTests.cs

```diff
@@ -437,8 +462,15 @@ private async Task ValidateResponseStream(AppRunner runner, Task<ResponseStreamH
 
             _outputHelper.WriteLine("Begin reading log entries.");
             string line;
+
             while (null != (line = await reader.ReadLineAsync()))
             {
+                if (format == LogFormat.Json)
+                {
+                    const char recordSeparator = '\u001E';
+                    line.Trim(recordSeparator);
```

Comment:
Let's be a bit more specific about this: validate that the first character is the record separator and the second character is not the record separator; then you can TrimStart. This will ensure that the beginning is exactly as expected and preserve the end.

Reply:
Would it also be useful to confirm that there is a second character? If not, I believe the line would be null once we TrimStart, in which case the while loop should break. Also, what behavior should we have if the validation fails (either the first character isn't the record separator or the second character is the record separator)? Should this be silently skipped over, should this be logged, or should we exit from the while loop due to the unexpected formatting?

Reply:
Let's be diligent by not assuming anything is valid and verify it. So something like:

```
Assert.True(line.Length > 1);
Assert.Equal(recordSeperator, line[0]);
Assert.NotEqual(recordSeperator, line[1]);
line = line.TrimStart(recordSeperator);
```

The Assert.* methods will throw an exception and end the test if the assertion fails.